tests/components/manual_mqtt/test_alarm_control_panel.py

```diff
@@ -11,1772 +10,1698 @@
-            ),
-            return_value=future,
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_ARMED_HOME, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
 
-        # Arm in away mode
-        common.alarm_arm_away(self.hass)
-        self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_PENDING, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
-        # Fast-forward a little bit
-        future = dt_util.utcnow() + timedelta(seconds=1)
-        with patch(
-            (
-                "homeassistant.components.manual_mqtt.alarm_control_panel."
-                "dt_util.utcnow"
-            ),
-            return_value=future,
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_ARMED_AWAY, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
-
-        # Arm in night mode
-        common.alarm_arm_night(self.hass)
-        self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_PENDING, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
-        # Fast-forward a little bit
-        future = dt_util.utcnow() + timedelta(seconds=1)
-        with patch(
-            (
-                "homeassistant.components.manual_mqtt.alarm_control_panel."
-                "dt_util.utcnow"
-            ),
-            return_value=future,
-        ):
-            fire_time_changed(self.hass, future)
-            self.hass.block_till_done()
-        self.mock_publish.async_publish.assert_called_once_with(
-            "alarm/state", STATE_ALARM_ARMED_NIGHT, 0, True
-        )
-        self.mock_publish.async_publish.reset_mock()
+async def test_arm_home_no_pending(hass, mqtt_mock):
+    """Test arm home method."""
+    assert await async_setup_component(
+        hass,
+        alarm_control_panel.DOMAIN,
+        {
+            "alarm_control_panel": {
+                "platform": "manual_mqtt",
+                "name": "test",
+                "code": CODE,
+                "pending_time": 0,
+                "disarm_after_trigger": False,
+                "command_topic": "alarm/command",
+                "state_topic": "alarm/state",
+            }
+        },
+    )
+    await hass.async_block_till_done()
+
+    entity_id = "alarm_control_panel.test"
+
+    assert STATE_ALARM_DISARMED == hass.states.get(entity_id).state
+
+    await common.async_alarm_arm_home(hass, CODE)
+    await hass.async_block_till_done()
+
+    assert STATE_ALARM_ARMED_HOME == hass.states.get(entity_id).state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)